lib/server.js

```diff
@@ -353,6 +353,15 @@ module.exports = class Server extends EventEmitter {
 
   _createServer() {
     this._httpServer = this._createHttpServer();
+
+    if (this.opts.http2 || this.opts.secure) {
```

Comment:
You can check only `this.opts.secure`, the `http2` options forces that.

Reply:
fixed :+1: